### DISCUSSION TITLE
Error making a rout in reverse

### DISCUSSION THREAD

**Discussion by CariocoSuzuki** [osmandapp/OsmAnd#18792] [Maps Related]
I have a predefined route in one direction and when I tell Osmand to carry out the "Reverse Route" option at the roundabouts it maintains the crossing points, i.e. if in the initial direction it passes on the right in reverse mode, the track passes on the left. I think that this is not convenient

**Comment by DmitryAlexei** [comment]
After changing the route direction please use "Attach to the roads" option.
For more information see https://osmand.net/docs/user/navigation/setup/gpx-navigation#attach-to-roads

**Comment by Sonwon1** [comment]
> After changing the route direction please use "Attach to the roads" option. For more information see https://osmand.net/docs/user/navigation/setup/gpx-navigation#attach-to-roads

In some cases the "Attach to the roads" option is so slow it is unusable. I wrote about this when importing a 300 km GPX file. Or has something changed in the 4.6 version? If it is unfixable then the documentation should say that it is slow and provide suggested workarounds when possible.

**Reply by DmitryAlexei** [reply]
The speed of attaching to the roads depends on how long your track is and the threshold distance. If the distance set to >50 m (e.g. 100m) the track calculates faster. I believe that for 300 km tracks, attaching may take a while. (from the link above ` If your track consists of a big amount of points, attaching to the roads may take a long time `).
I suppose, that after the new routing mechanism is fully implemented, attaching to roads will be quicker.
